### PR TITLE
Simplify IndexCoord logic

### PR DESCRIPTION
**Pull request by xiaocai2333**
Signed-off-by: cai.zhang <cai.zhang@zilliz.com>
issue: #21269

### CODE REVIEW

File: scripts/start_cluster.sh [hunk 1]
```diff
@@ -35,6 +35,9 @@ nohup ./bin/milvus run datacoord > /tmp/datacoord.log 2>&1 &
 echo "Starting datanode..."
 nohup ./bin/milvus run datanode > /tmp/datanode.log 2>&1 &
 
+echo "Starting indexnode..."
+nohup ./bin/milvus run indexnode > /tmp/indexnode.log 2>&1 &
```
Comment (czs007) [inline scripts/start_cluster.sh:39]: just remove this

File: cmd/components/index_coord.go [hunk 2]
```diff
@@ -20,56 +20,32 @@ import (
...
 	return nil
 }
 
 // GetComponentStates returns indexnode's states
 func (s *IndexCoord) Health(ctx context.Context) commonpb.StateCode {
-	resp, err := s.svr.GetComponentStates(ctx, &milvuspb.GetComponentStatesRequest{})
-	if err != nil {
-		return commonpb.StateCode_Abnormal
-	}
-	return resp.State.GetStateCode()
```
Comment (czs007) [inline cmd/components/index_coord.go:72]: add comment

File: internal/proto/index_coord.proto [hunk 3]
```diff
@@ -8,26 +8,6 @@ import "common.proto";
...
-  rpc GetIndexBuildProgress(GetIndexBuildProgressRequest) returns (GetIndexBuildProgressResponse) {}
-
-  rpc ShowConfigurations(internal.ShowConfigurationsRequest) returns (internal.ShowConfigurationsResponse){}
-  // https://wiki.lfaidata.foundation/display/MIL/MEP+8+--+Add+metrics+for+proxy
-  rpc GetMetrics(milvus.GetMetricsRequest) returns (milvus.GetMetricsResponse) {}
-
-  rpc CheckHealth(milvus.CheckHealthRequest) returns (milvus.CheckHealthResponse) {}
-}
-
 service IndexNode {
```
[the remark this answers is not in the document]
Comment (xiaocai2333) [reply internal/proto/index_coord.proto:11]: ok

**Pull request merged**